lrpar/src/lib/ctbuilder.rs

```diff
@@ -357,7 +357,40 @@ where
         }
 
         let inc = read_to_string(grmp).unwrap();
-        let grm = YaccGrammar::<StorageT>::new_with_storaget(yk, &inc)?;
+
+        fn off_to_line_col(src: &str, off: usize) -> (usize, usize) {
+            let newlines: Vec<usize> = src
+                .char_indices()
+                .peekable()
+                .filter_map(|c| match c {
+                    (offset, '\n') | (offset, '\r') => Some(offset),
+                    _ => None,
+                })
+                .collect::<Vec<_>>();
+            if off == src.len() {
+                let line_off = newlines.last().unwrap();
+                return (newlines.len(), src[*line_off..].chars().count() + 1);
+            }
+            let (line_m1, &line_off) = newlines
+                .iter()
+                .enumerate()
+                .rev()
+                .find(|&(_, &line_off)| line_off <= off)
+                .unwrap();
+            let c_off = src[line_off..]
+                .char_indices()
+                .position(|(c_off, _)| c_off == off - line_off)
+                .unwrap();
+            (line_m1 + 1, c_off + 1)
+        }
+
+        let grm = YaccGrammar::<StorageT>::new_with_storaget(yk, &inc).map_err(|e| match e {
+            YaccGrammarError::YaccParserError(e) => {
+                let (column, line) = off_to_line_col(&inc, e.span.start());
+                format!("{} at column {column} line {line}", e)
+            }
+            e => e.to_string(),
+        })?;
```

Comment:
This is a bit hacky going from `YaccGrammarError -> String -> Box<dyn Error>`, but I'm not so sure what we could actually do otherwise